### PR TITLE
Update Gtest

### PR DESCRIPTION
**Pull request by sslattery**
Uses the gtest method of pulling a tagged version instead of including a snapshot in the repo.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -9,6 +9,23 @@ cmake_minimum_required(VERSION 3.9)
 project(Cabana LANGUAGES CXX)
 set(PROJECT_VERSION "0.3.0-dev")
 
+# Download and unpack googletest at configure time
+configure_file(CMakeLists.txt.in googletest-download/CMakeLists.txt)
+execute_process(COMMAND "${CMAKE_COMMAND}" -G "${CMAKE_GENERATOR}" .
+    WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/googletest-download" )
+execute_process(COMMAND "${CMAKE_COMMAND}" --build .
+    WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/googletest-download" )
```
Comment (junghans) [inline CMakeLists.txt]: This seems to be a hackery to make is work with older versions of cmake, I think we should require cmake-3.11 and use [FetchContent](https://cmake.org/cmake/help/v3.11/module/FetchContent.html) instead:
```
FetchContent_Declare(
  googletest
  GIT_REPOSITORY https://github.com/google/googletest.git
  GIT_TAG        release-1.10.0
)

FetchContent_GetProperties(googletest)
if(NOT googletest_POPULATED)
  FetchContent_Populate(googletest)
  add_subdirectory(${googletest_SOURCE_DIR} ${googletest_BINARY_DIR})
endif()
```

Comment (sslattery) [reply CMakeLists.txt]: I'm fine with that - I just basically copy/pasted what was in the GTest manual

Comment (sslattery) [reply CMakeLists.txt]: so no `CMakeLists.txt.in` then?

Comment (junghans) [reply CMakeLists.txt]: Yeah.

Comment (junghans) [reply CMakeLists.txt]: My code snippet above is actually taken for the CMake manual ;-)

Comment (sslattery) [reply CMakeLists.txt]: ok - fixed

File: CMakeLists.txt.in [hunk 2]
```diff
@@ -0,0 +1,15 @@
+cmake_minimum_required(VERSION 2.8.11)
```
Comment (junghans) [inline CMakeLists.txt.in]: I think that file can be removed now.

Comment (sslattery) [reply CMakeLists.txt.in]: do we need to increase the minimum version then?

File: CMakeLists.txt.in [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+project(googletest-download NONE)
```
Comment (dalg24) [inline CMakeLists.txt.in]: Why do we not rely on `ExternalProject` to download the source?

Comment (sslattery) [reply CMakeLists.txt.in]: That's how we did it in Cajita originally but @junghans suggested we use `FetchContent` instead

File: CMakeLists.txt.in [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+ExternalProject_Add(googletest
+    GIT_REPOSITORY https://github.com/google/googletest.git
+    GIT_TAG release-1.10.0
+    SOURCE_DIR "${CMAKE_BINARY_DIR}/googletest-src"
+    BINARY_DIR "${CMAKE_BINARY_DIR}/googletest-build"
+    CONFIGURE_COMMAND ""
+    BUILD_COMMAND ""
+    INSTALL_COMMAND ""
+    TEST_COMMAND ""
+)
```
Comment (dalg24) [inline CMakeLists.txt.in]: Wait what does that even do if you comment out all steps??

File: core/performance_test/peakflops/CMakeLists.txt [hunk 5]
```diff
@@ -3,6 +3,13 @@ if(NOT Cabana_ENABLE_TESTING)
   return()
 endif()
 
+include_directories(${CMAKE_CURRENT_BINARY_DIR})
+include_directories(${CMAKE_CURRENT_SOURCE_DIR})
+include_directories(${CMAKE_PROJECT_BINARY_DIR}/include)
```
Comment (dalg24) [inline core/performance_test/peakflops/CMakeLists.txt]: Why would you need to add that to the include path?

File: core/performance_test/peakflops/CMakeLists.txt [hunk 6]
```diff
@@ -3,6 +3,13 @@ if(NOT Cabana_ENABLE_TESTING)
...
+include_directories(${CMAKE_PROJECT_BINARY_DIR}/googletest-src/googletest/include)
```
Comment (dalg24) [inline core/performance_test/peakflops/CMakeLists.txt]: You link against the `gtest` target.

File: core/unit_test/CMakeLists.txt [hunk 7]
```diff
@@ -3,19 +3,10 @@ if(NOT Cabana_ENABLE_TESTING)
   return()
 endif()
 
-set(GTEST_SOURCE_DIR ${CMAKE_SOURCE_DIR}/gtest)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DGTEST_HAS_PTHREAD=0")
```
Comment (dalg24) [inline core/unit_test/CMakeLists.txt]: Do we not need to set `GTEST_HAS_PTHREAD` any more?

File: CMakeLists.txt [hunk 8]
```diff
@@ -1,14 +1,30 @@
...
 # Define your project name and set up major project options
 ##---------------------------------------------------------------------------##
 project(Cabana LANGUAGES CXX)
 set(PROJECT_VERSION "0.3.0-dev")
 
+# Download and unpack googletest at configure time
+include(FetchContent)
+FetchContent_Declare(
+  googletest
+  URL https://github.com/google/googletest/archive/release-1.10.0.tar.gz
```
Comment (streeve) [inline CMakeLists.txt]: The URL download fails on the LLNL machines I've tried where the Git repo has worked fine. Not sure if there's something I can change on my end

File: core/performance_test/peakflops/CMakeLists.txt [hunk 9]
```diff
@@ -3,6 +3,9 @@ if(NOT Cabana_ENABLE_TESTING)
   return()
 endif()
 
+set(gtest_args --gtest_color=yes)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DGTEST_HAS_PTHREAD=0")
```
Comment (junghans) [inline core/performance_test/peakflops/CMakeLists.txt]: Related to #149, we should replace this in favor of `target_compile_definitions(<target> PRIVATE GTEST_HAS_PTHREAD=0)` in the loop below.

File: core/unit_test/CMakeLists.txt [hunk 10]
```diff
@@ -53,8 +40,8 @@ macro(Cabana_add_tests)
         file(APPEND ${_file} "#include <tst${_test}.hpp>\n")
         set(_target ${_test}_test_${_device})
         add_executable(${_target} ${_file} ${CABANA_UNIT_TEST_MAIN})
-        target_include_directories(${_target} PUBLIC ${_dir})
-        target_link_libraries(${_target} cabanacore cabana_core_gtest)
+        target_include_directories(${_target} PUBLIC ${_dir} ${CMAKE_CURRENT_SOURCE_DIR})
+        target_link_libraries(${_target} cabanacore gtest)
         if(CABANA_UNIT_TEST_MPI)
           foreach(_np ${CABANA_UNIT_TEST_MPIEXEC_NUMPROCS})
             # NOTE: When moving to CMake 3.10+ make sure to use MPIEXEC_EXECUTABLE instead
```
[the remark this answers is not in the document]
Comment (junghans) [reply core/unit_test/CMakeLists.txt]: Good find.

File: CMakeLists.txt (hunk 8 above)
Comment (sslattery) [reply CMakeLists.txt]: We can move to `git` but will need to update our Docker image - @dalg24 thoughts?

File: CMakeLists.txt [hunk 11]
```diff
@@ -1,14 +1,30 @@
...
 # Define your project name and set up major project options
 ##---------------------------------------------------------------------------##
 project(Cabana LANGUAGES CXX)
 set(PROJECT_VERSION "0.3.0-dev")
 
+# Download and unpack googletest at configure time
+include(FetchContent)
+FetchContent_Declare(
+  googletest
+  URL https://github.com/google/googletest/archive/release-1.10.0.tar.gz http://distfiles.macports.org/gtest/googletest-1.10.0.tar.gz
```
Comment (junghans) [inline CMakeLists.txt]: We should add `URL_HASH`.

File: CMakeLists.txt (hunk 8 above)
Comment (junghans) [reply CMakeLists.txt]: @dalg24 does the jenkins machine have no internet connection?

File: core/unit_test/CMakeLists.txt (hunk 7 above)
Comment (sslattery) [reply core/unit_test/CMakeLists.txt]: I don't think so - that was a holdover from a failing Mac build which is working now

Comment (sslattery) [reply core/unit_test/CMakeLists.txt]: I don't think so - that was a holdover from a failing Mac build which is working now

**Comment by sslattery** [comment]
This is lingering and we don't have a good fix. I'm fine snapshotting in the Gtest that works for us for now.

**Pull request closed without merging**